src/layering.jl

```diff
@@ -3,7 +3,8 @@
 function layer_by_longest_path_to_source(graph, force_layer)
     dists = longest_paths(graph, sources(graph))
     force_layers!(graph, dists, force_layer)
-    layer_groups = collect.(IterTools.groupby(i->dists[i], sort(vertices(graph), by=i->dists[i])))
+    layer_groups = [findall(==(d), dists) for d in 1:maximum(dists)]
```

Comment:
Note the different time complexity.
This is  `O(n*d)` where as  the previous code is `O(n* log(n))`

But it seems we can do better with 

```suggestion
    layer_groups = [Int[] for d in 1:maximum(dists)]
    for (ii, d) in enumerate()
        push!(layer_groups[d], ii)
    end
```

Which is only `O(n + d)`

Reply:
Ah, I hadn't thought this. Your suggestion is very nice: I will add that now 👍 !

Reply:
< Would it be more natural to simply make this a Dict{Int, Vector{Int}} instead? Then there could also be negative indices and we wouldn't have to potentially store a bunch of empty layers.

My instinct is that in practical cases the extra costs from empty layers will be lower than the extra cost of needing to do a hash.

Where would we get negative layer indexes? Would that be just from users specifying them by hand?
So it would be a way to say to put this **before** the natural root layer?
That sounds like a bigger change that would want to be in its own PR


> Maybe there's some assumptions in the code elsewhere that assumes iteration over layer2nodes in incrementing fashion?

Good question. I don't think so, but i can't from memory rule it out

Reply:
Agree, better to leave that for a potential follow-up, if ever relevant. Thanks for expeditious merging 👍 !